### PR TITLE
Add timeseries page to dashboard

### LINKED ISSUE #568: Add timeseries functionality to dashboard
See lime (and rise) notebooks in tutorials

https://github.com/dianna-ai/dianna/blob/main/tutorials/rise_timeseries_weather.ipynb
https://github.com/dianna-ai/dianna/blob/main/tutorials/lime_timeseries_weather.ipynb

### PR DESCRIPTION
**Pull request by stefsmeets**
This PR adds a page for timeseries to the dashboard.

Closes #568

### Todo

- [x] Add caching
- [x] What is the file type/format for data uploads?
- [x] Update lime interface

### CODE REVIEW

**Comment by stefsmeets** [comment]
Hi @geek-yang @cwmeijer , is either of you interested in reviewing this PR?

File: dianna/methods/lime_timeseries.py [hunk 1]
```diff
@@ -117,15 +116,15 @@ def explain(
...
+        for i, label in enumerate(labels):
+            local_exp = sorted(exp.local_exp[label])
             # shape of local_exp [(index, saliency)]
             selected_saliency = [i[1] for i in local_exp]
             saliency.append(selected_saliency[:])
 
         return np.concatenate(saliency).reshape(-1, sequence, n_var)
 
-    def _calculate_distance(self, masked_data, distance_method="cosine"):
+    def _calculate_distance(self, masked_data, distance_method='cosine'):
```
Comment (geek-yang) [inline dianna/methods/lime_timeseries.py:127]: Just notice that your formatter prefers single quote. I use `black` and it turns everything to double quote. Maybe we can create an issue and put it in the developer doc about which formatter we prefer?

**Review by geek-yang — APPROVED** [review]
I went over the changes roughly and most of them look pretty nice to me (ps: don't have much experience with `streamlit`. But I think the code itself looks pretty and very well-organized. It is very readable. I could follow the code and figure it out what are most of the functions used for. Nice lesson learned for me about `streamlit` 😄 ).

Just some small comments to address, before merging it to main branch. But I can give you a go already.

I put more time on testing the dashboard, especially for timeseries panel. It looks pretty awesome. And the added cache functionality also works for me perfectly. Thanks a lot for the fabulous work 👍!

**Comment by elboyran** [comment]
[heart]         Elena Ranguelova reacted to your message:
________________________________

File: dianna/methods/lime_timeseries.py [hunk 2]
```diff
@@ -117,15 +116,15 @@ def explain(
             )
         # extract scores from lime explainer
         saliency = []
-        for i, _ in enumerate(labels):
-            local_exp = sorted(exp.local_exp[i])
+        for i, label in enumerate(labels):
+            local_exp = sorted(exp.local_exp[label])
```
[the remark this answers is not in the document]
Comment (stefsmeets) [reply dianna/methods/lime_timeseries.py:120]: For me the original code caused it to crash. If the labels are strings like we have in the dashboard, then the lookup in `exp.local_exp` raises an exception.

File: dianna/methods/lime_timeseries.py (hunk 1 above)
Comment (stefsmeets) [reply dianna/methods/lime_timeseries.py:127]: I'm pretty sure we already have this: https://dianna.readthedocs.io/en/latest/developer_info.html#running-linters-locally 😅

File: dianna/methods/lime_timeseries.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (geek-yang) [reply dianna/methods/lime_timeseries.py:120]: That's strange. Actually I haven't tested it with strings. But even with strings, the index should always work. Anyway, thanks for noticing it. If the current code works for dashboard, let's keep it for now.

**Comment by geek-yang** [comment]
> > I got a small concern about sonarcloud check. If we merge it as what it is for now, the coverage will drop dramatically. Since we don't want to add unit tests for each function in streamlit (also it makes no sense to do so at this point). Shall we just skip `dashboard` folder only for sonarcloud check?
> 
> Much of this code is still being tested, just not taken into account for coverage.
> 
> The issue is that the code is run in a subprocess, which cannot be measured automatically for coverage. This would be the way to go about it: https://coverage.readthedocs.io/en/stable/subprocess.html#configuring-python-for-sub-process-measurement

I saw the tests with `playwright` but didn't realize that it already covers most of the stuff, while they were merely not reflected by `sonarcloud`. Thanks for clarifying it. The solution you propose looks good to me. Let's give a try then.

**Comment by stefsmeets** [comment]
> > > I got a small concern about sonarcloud check. If we merge it as what it is for now, the coverage will drop dramatically. Since we don't want to add unit tests for each function in streamlit (also it makes no sense to do so at this point). Shall we just skip `dashboard` folder only for sonarcloud check?
> > 
> > 
> > Much of this code is still being tested, just not taken into account for coverage.
> > The issue is that the code is run in a subprocess, which cannot be measured automatically for coverage. This would be the way to go about it: https://coverage.readthedocs.io/en/stable/subprocess.html#configuring-python-for-sub-process-measurement
> 
> I saw the tests with `playwright` but didn't realize that it already covers most of the stuff, while they were merely not reflected by `sonarcloud`. Thanks for clarifying it. The solution you propose looks good to me. Let's give a try then.

Let's follow up later then. I spent the last hour fighting this, but I can't get it to work at the moment.

**Pull request merged**